scrapi/harvesters/vivo.py

```diff
@@ -0,0 +1,216 @@
+                                        ?URI vivo:freetextKeyword ?keyword .
+                                    }}
+                            OPTIONAL {{
+                                        ?URI vitro:mostSpecificType ?typeURI .
+                                        ?typeURI rdfs:label ?type .
+                                    }}
+                            OPTIONAL {{
+                                        ?URI bibo:abstract ?abstract .
+                                    }}        
+                            OPTIONAL {{
+                                        ?URI bibo:doi ?doi .
+                                    }}
+                            OPTIONAL {{
+                                        ?URI bibo:volume ?volume .
+                                    }}
+                            OPTIONAL {{
+                                        ?URI bibo:number ?number .
+                                    }}
+                            OPTIONAL {{
+                                        ?URI bibo:pageStart ?startPage .
+                                    }}
+                            OPTIONAL {{
+                                        ?URI bibo:pageEnd ?endPage .
+                                    }}
+                            OPTIONAL {{
+                                        ?URI bibo:pmid ?PMID .
+                                   }} 
+                            OPTIONAL {{
+                                        ?autorship a vivo:Authorship .
+                                        ?autorship vivo:relates ?URI .
+                                        ?authorURI vivo:relatedBy ?autorship .
+                                        ?authorURI a foaf:Person .
+                                        ?authorURI rdfs:label ?authorName .
+                                        BIND ( CONCAT(str(?authorURI), ';' , ?authorName) AS ?author) .
+                                    }}
+                            FILTER (strdt(?date, xsd:date) >= "{}"^^xsd:date && strdt(?date, xsd:date) <= "{}"^^xsd:date) 
+                        }} GROUP BY ?URI ?doi ?abstract ?title ?journalTitle ?date ?publisher ?volume ?number ?startPage ?endPage ?PMID ?ISSN ?ISBN LIMIT {} OFFSET {}
+                    """
+        query_str = query_str.format(start_date.isoformat(), end_date.isoformat(), limit, offset)
+        self.sparql_wrapper.setQuery(query_str)
+        results = self.sparql_wrapper.query()
+        results = results.convert()
+        return results['results']['bindings']
+
+    @property
+    def schema(self):
+        return {
+            'title': ('/title', lambda x: x if x else ''),
+            'description': ('/subtitle', lambda x: x[0] if (isinstance(x, list) and x) else x or ''),
+            'providerUpdatedDateTime': ('/date', lambda x: datetime.strptime(x, "%Y-%m-%d").strftime("%Y-%m-%dT%H:%M:%S%Z")+'+00:00'),
+            'uris': {
+                'canonicalUri': '/URI'
```

Comment:
canonicalUri actually should be duplicated to one of the other uri fields as well (either `uris.objectUris`, `uris.providerUris`, or `uris.descriptorUris`, depending one what it points to)